vcr/api/v2/holder.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Nuts node
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package v2
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/vcr/concept"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+)
+
+// typeHelper helps in transforming the given JSON-LD document to the correct query
+// If it contains NutsOrganizationCredential then we need to search for an organization
+// NutsAuthorizationCrerdential for authorization credentials
+type typeHelper struct {
+	Type []string `json:"type"`
+}
+
+// searchType returns the first type that is in this list:
+// - NutsOrganizationCredential
+// - NutsAuthorizationCredential
+func (ch typeHelper) searchType() string {
+	for _, c := range ch.Type {
+		switch c {
+		case credential.NutsOrganizationCredentialType:
+			fallthrough
+		case credential.NutsAuthorizationCredentialType:
+			return c
+		}
+	}
+	return ""
+}
+
+// SearchVCs checks the context used in the JSON-LD query, based on the contents it maps to a non-JSON-LD query
+// After V1, this needs to be remapped to a DB search that supports native JSON-LD
+func (w *Wrapper) SearchVCs(ctx echo.Context, params SearchVCsParams) error {
+	var rawRequest typeHelper
```

Comment:
it's OK, but just could even be an anonymous type inside this function! (since it's only used here).